### ISSUE TITLE
Can you list the steps of porting lwip step by step ? 

### ISSUE THREAD

**Issue by bootbool** [michael-scharf/lwip-echoserver#1]
Hello, I try to port to original lwip code to linux step by step , to archive the same effect of your work, but,  lwip does not reply any packet. 
Neither can I  get any guide from the commit history. Wish you give me some key advises to do the work.

**Comment by michael-scharf** [comment]
Sorry, I don't have such a step-by-step list. It took some trial-and-error attempts until I had a working user-space echo-server solution.

In order to have an example that is easy to understand (for students), I have removed parts of the lwip code that are not needed for the echo-server, most notably the other apps and ports other than "unix". I have also replaced the cmake build tool chain of lwip by a simple Makefile. That required, amongst others, some new symlinks in include/. So, there are some changes to the upstream repository, and a diff to the original lwip repository doesn't make a lot of sense.

Note that the build system and maybe some few files in src/ and include/ are taken from https://github.com/heiher/lwip, as I didn't figure out how to set up an echo server just using the lwip and lwip contrib code. The resulting echo-server example in this repository should work out-of-the-box. At least it does so on Debian 12.

Upgrade of this repository to a newer version of lwip shouldn't be difficult. The content of src/ and include/ is just taken from the upstream for all those files that are part of the lwip release. 

So, for an upgrade of the echo-server to a newer lwip release, it should be sufficient to replace the content of the src/ and include/ folders to the more recent version of the same directories in the lwip upstream. Maybe some minor fixes could be required in the files not taken from lwip, but IMHO it should be straightforward.
